ssl/quic/quic_wire_pkt.c

```diff
@@ -451,6 +457,12 @@ int ossl_quic_wire_decode_pkt_hdr(PACKET *pkt,
         }
     }
 
+    /*
+     * Good decode, clear the generic DECODE_ERR flag
+     */
+    if (fail_cause != NULL)
```

Comment:
I think we should also check a packet length here. RFC 9000, section 6
```
   The size of the first packet sent by a client will determine whether
   a server sends a Version Negotiation packet.  Clients that support
   multiple QUIC versions SHOULD ensure that the first UDP datagram they
   send is sized to the largest of the minimum datagram sizes from all
   versions they support, using PADDING frames (Section 19.1) as
   necessary.  This ensures that the server responds if there is a
   mutually supported version.  A server might not send a Version
   Negotiation packet if the datagram it receives is smaller than the
   minimum size specified in a different version; see Section 14.1.
```

section 14.1 says:
```
   A server MUST discard an Initial packet that is carried in a UDP
   datagram with a payload that is smaller than the smallest allowed
   maximum datagram size of 1200 bytes.  A server MAY also immediately
   close the connection by sending a CONNECTION_CLOSE frame with an
   error code of PROTOCOL_VIOLATION; see Section 10.2.3.
```
If I understand the idea right the server should not start talking/replying to client if the first packet sent by client is less than 1200, those packets are suspicious and should be discarded. So serious client if starts with negotiation should add a padding of 1200 bytes. so if `l < 1200` here, we should just keep `DECODE_ERR` here, so we won't be replying with protocol version negotiation.